client/db/bolt/db.go

```diff
@@ -298,9 +299,30 @@ func (db *BoltDB) CreateAccount(ai *dexdb.AccountInfo) error {
 	})
 }
 
+// Make account info key from client pub key
+func accountInfoKey(accountID account.AccountID) []byte {
+	acctKey := make([]byte, len(accountID))
+	copy(acctKey, accountID.String())
+	return acctKey
+}
+
+// DisableAccount sets account info disabled flag to true
+func (db *BoltDB) DisableAccount(ai *dexdb.AccountInfo) error {
+	acctKey := accountInfoKey(ai.ID)
+	return db.acctsUpdate(func(accts *bbolt.Bucket) error {
+		acct := accts.Bucket(acctKey)
+		if acct == nil {
+			return fmt.Errorf("account not found for %s", ai.Host)
+		}
+		ai.Disabled = true
+		err := acct.Put(accountKey, ai.Encode())
+		return err
+	})
+}
+
 // AccountPaid marks the account as paid by setting the "fee proof".
-func (db *BoltDB) AccountPaid(proof *dexdb.AccountProof) error {
-	acctKey := []byte(proof.Host)
+func (db *BoltDB) AccountPaid(proof *dexdb.AccountProof, ai *dexdb.AccountInfo) error {
+	acctKey := accountInfoKey(ai.ID)
```

Comment:
How about adding an `account.AccountID` field to `dexdb.AccountProof` instead of passing another argument to `AccountPaid`?

Reply:
If it's just for the account ID, I'd rather it was just passed as the first argument.

`AccountPaid(acctID account.AccountID, proof *dexdb.AccountProof)`

But really, I don't think we should key on account ID to begin with. `EncKey` is just as unique, and using account ID requires either **a)** the caller to decrypt the private key, or **b)** the public key to be stored separately.